### PR TITLE
macdependency: new port, version 1.1.0

### PR DESCRIPTION
**Pull request by GiovanniBussi**
#### Description

New port submission. 

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.11.6 15G20015
Xcode 8.1 8B62 

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by GiovanniBussi** [comment]
@l2dy I changed the commit message, amended and force pushed. Thanks!

File: devel/MacDependency/Portfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+platforms           darwin
+
+checksums           rmd160  071c900c94d61f6796125b667df7a67e85daa91f \
+                    sha256  83aab373efefa4e0f1b1b9653d21ab6702924ac0202f09b0c2b73faf3ae01cf9 \
+                    size    1268109
+
+xcode.project       MacDependency/MacDependency.xcodeproj
+
+# framework
+subport MachO {
```
Comment (l2dy) [inline devel/MacDependency/Portfile]: This name is a bit too generic, perhaps `MacDependency-MachO` is better?

File: devel/MacDependency/Portfile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    xcode.destroot.type framework
+}
+
+# main app
+if {${name} eq ${subport}} {
+    xcode.build.settings FRAMEWORK_SEARCH_PATHS=${frameworks_dir}
+    xcode.destroot.settings FRAMEWORK_SEARCH_PATHS=${frameworks_dir}
+    depends_lib-append port:MachO
+}
+
```
Comment (l2dy) [inline devel/MacDependency/Portfile]: Please remove extra newline at EOF.

**Comment by l2dy** [comment]
Can you show me the output of `port contents MacDependency` and `port contents MacDependency-MachO`? From the logs, the `MacDependency` port seems to contain the framework.

**Comment by GiovanniBussi** [comment]
@l2dy you are right I think there is some problem in the way I configured it.

I will try to sort it out and be back when I am done.

Sorry for that, I am not very familiar with xcode build system.

Giovanni

**Comment by pmetzger** [comment]
@GiovanniBussi Any progress?

**Comment by GiovanniBussi** [comment]
@pmetzger not yet sorry.... If you prefer I can close the pull request and reopen it later when I will have been able to fix the issue. Though I am afraid I might be not familiar enough with `xcodebuild` to solve the problem myself.

**Comment by kencu** [comment]
Perhaps the subport is not needed here -- what happens if you just delete the subport and install the app?

**Comment by GiovanniBussi** [comment]
Hi!

If I install the app only (after having uninstalled both ports) I get this error
````
:info:build ** BUILD FAILED **
:info:build The following build commands failed:
:info:build     CompileC build/MacDependency.build/Release/MacDependency.build/Objects-normal/x86_64/SymbolTableEntryModel.o                      SymbolTableEntryModel.mm normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler
:info:build (1 failure)
:info:build Command failed: cd "/opt/local/var/macports/build/_Users_bussi_ports_devel_MacDependency/MacDependency/work/macdependency-1.1.0" && / usr/bin/xcodebuild -project "MacDependency/MacDependency.xcodeproj" -alltargets -configuration Deployment build INSTALL_PATH=/Applications/       MacPorts OBJROOT=build/ SYMROOT=build/ MACOSX_DEPLOYMENT_TARGET=10.11 ARCHS=x86_64 SDKROOT="" GCC_VERSION=com.apple.compilers.llvm.clang.1_0      CLANG_CXX_LIBRARY="libc++" FRAMEWORK_SEARCH_PATHS=/opt/local/Library/Frameworks
:info:build Exit code: 65
:error:build Failed to build MacDependency: command execution failed
````

On the other hand, if I install first the framework and then the app things work. That's the reason why I added the framework as a build dependency, though I might have misinterpreted something.

Thanks!

Giovanni

**Comment by pmetzger** [comment]
Okay, so where are we now? If this isn't ready, @GiovanniBussi, do indeed close it and re-open it when you have time.

**Comment by GiovanniBussi** [comment]
@pmetzger ok I close it right now. Thanks anyway!

**Comment by kencu** [comment]
Sounds like the build of the app didn't know how to find the not-yet-installed framework during construction. Perhaps adding a framework search path during app construction might fix that...

**Pull request closed without merging**